pkg/register/hook.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2017 by the contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package register
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/github"
+)
+
+func hasSignOffHook(gh *github.Client, org string, repo *github.Repository, url string) (bool, error) {
+	opt := &github.ListOptions{PerPage: 10}
+	for {
+		hooks, resp, err := gh.Repositories.ListHooks(context.TODO(), org, repo.GetName(), opt)
+		if resp != nil && resp.StatusCode == 404 {
+			// 404 just means there are no hooks for this repo
+			return false, nil
+		}
+		if err != nil {
+			return false, fmt.Errorf("Error listing hooks for %q: %v", repo.GetFullName(), err)
+		}
+		for _, hook := range hooks {
+			// if the hook with our expected URL already exists, we're done
+			if hook.Config["url"] == url {
```

Comment:
Hmm -- the secret may be wrong.  But I'm guessing that this doesn't return the secret.  At some point we may want to have a one shot mode to force re-registration in the event we rotate the secret.  That might be a good roadmap/TODO thing.

Reply:
Yeah, you can't really validate that the secret matches. You just have to overwrite it if you want to make sure it's correct. I'll file an issue for the force resync command.